plugin/src/main/java/io/fabric8/maven/plugin/SetupMojo.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2005-2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.plugin;
+
+import io.fabric8.maven.core.util.MavenUtil;
+import io.fabric8.utils.DomHelper;
+import io.fabric8.utils.Strings;
+import io.fabric8.utils.XmlUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Text;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+
+import static io.fabric8.utils.DomHelper.firstChild;
+
+/**
+ * Ensures that the fabric8 maven plugin is defined and setup correctly in the projects pom.xml
+ */
+@Mojo(name = "setup", requiresProject = true)
+public class SetupMojo extends AbstractFabric8Mojo {
+
+    public static final String PLUGIN_GROUPID = "io.fabric8";
+    public static final String PLUGIN_ARTIFACTID = "fabric8-maven-plugin";
+    public static final String FABRIC8_MAVEN_PLUGIN_VERSION_PROPERTY = "fabric8.maven.plugin.version";
+
+    /**
+     * Should we update the version if the fabric8-maven-plugin is already used on a project?
+     */
+    @Parameter(property = "updateVersion", defaultValue = "true")
```

Comment:
I see your point. I'm somewhat in between simplicity (with possible clashes to other plugins, you still remember `docker.image` ;-) ?) and consistency (we _always_ prefix with `docker.` and `fabric8.` for our stuff). 

But tbh, personally I never would use an update because I would be afraid that things are messed up. Just thought about ppl using profiles, too. Do we update these as well ? Sorry, for me _update_ still feels like Pandora's box.

However, if you really insist on going this route, maybe we could set the `update` property by default to false (to avoid accidentally mess up of their pom.xml) ?